handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -1557,14 +1559,26 @@ public final synchronized void setSSLParameters(SSLParameters sslParameters) {
 
         int version = PlatformDependent.javaVersion();
         if (version >= 7) {
-            endPointIdentificationAlgorithm = sslParameters.getEndpointIdentificationAlgorithm();
+            final String endPointIdentificationAlgorithm = sslParameters.getEndpointIdentificationAlgorithm();
+            final boolean endPointVerificationEnabled = endPointIdentificationAlgorithm != null &&
+                    !endPointIdentificationAlgorithm.isEmpty();
+            SSL.setHostNameValidation(ssl, DEFAULT_HOSTNAME_VERIFICATION_FLAGS,
+                    endPointVerificationEnabled ? getPeerHost() : null);
```

Comment:
I don't think we need to be this strict. These values are not strictly defined and are just used as strings. The API (referenced above) says if it is null/non-empty we should verify the certs.

I need to investigate the RFCs referenced by the docs for LDAPS and HTTPS to see if OpenSSL's options can mimic the behavior of the JDK.